test/unit/rules/no-nested-landmark-test.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const rule = require('../../../lib/rules/no-nested-landmark');
+
+const { createErrorMessage } = rule;
+
+generateRuleTests({
+  name: 'no-nested-landmark',
+
+  config: true,
+
+  good: [
+    '<div><main></main></div>',
+    '<div role="application"><div role="document"><div role="application"></div></div></div>',
+  ],
+
+  bad: [
```

Comment:
Can you add some additional combos of nesting that we want to prevent? All of these tests (at the moment) we are only checking `main` nested inside `main`.

Reply:
Since the rule is checking only for landmark elements, it will cover all the elements and roles defined here:
```js
const LANDMARK_ELEMENTS = ['header', 'main', 'aside', 'form', 'main', 'nav', 'footer'];
const ROLES = ['banner', 'main', 'complementary', 'form', 'search', 'navigation', 'contentinfo'];
```
Adding tests for all the items above will be overkill I guess. Nevertheless I tried to capture all the variations for a single element (main) with roles and nesting.
Added for `nav` and `header`, the most commonly used pattern AFAIK. Would be glad to add more if required.
What do you think? @MelSumner

Reply:
Ya, I agree we don't need a test for every combo. I was just looking for at least _one_ test for non-main.